src/query.rs

```diff
@@ -352,6 +395,15 @@ impl<'w, Q: Query> QueryBorrow<'w, Q> {
         unsafe { BatchedIter::new(self.meta, self.archetypes.iter(), batch_size) }
     }
 
+    /// Prepare this query for repeated execution
+    ///
+    /// A prepared query can be stored independently of the [world][World] and
+    /// is used to amortise the cost of the looking up the machting [archetypes][Archetype]
+    /// and the relevant type states therein.
+    pub fn prepare(self, world: &World) -> PreparedQuery<Q> {
```

Comment:
Having to pass in the same world twice in `world.query().prepare(&world)` is awkward. Since `PreparedQuery` has to support detecting and rebuilding stale state anyway, why don't we drop the `world` argument and instead construct the `PreparedQuery` in an initially stale state, so that it initializes itself on first use?

Reply:
Will change this, I just need to make sure that there exists a `memo` value that now valid `World` will have, probably by making the ID start at 1 instead of 0.

Reply:
Added this as a last commit. I did actually go a bit further and removed the `QueryBorrow::prepare` method and just gave `PreparedQuery` a `Default` implementation. Where I have applied this, we ended up storing the queries together with other system-internal state and often ended up initialising them using `Default` so I think this would be the most useful variant.

Reply:
Good idea!